registry/image.go

```diff
@@ -0,0 +1,128 @@
+package registry
+
+import (
+	"fmt"
+	"strings"
+	"time"
+)
+
+const (
+	dockerHubHost    = "index.docker.io"
+	dockerHubLibrary = "library"
+)
+
+// An image is a fully qualified name that refers to a particular image.
+// It is in the format: host[:port]/orgname/reponame[:tag]
+// Here, we refer to the "name" == orgname/reponame
+
+// Image can't really be a primitive string only, because we need to also
+// record information about it's creation time. (maybe more in the future)
+type Image interface {
+	WithTag(tag string) Image
+	WithCreatedAt(t *time.Time) Image
+	Components() (host, org, repo, tag string)
+	Host() string
+	Org() string
+	Repo() string
+	Tag() string
+	FQN() string
+	HostOrgRepo() string
+	OrgRepo() string
+	CreatedAt() *time.Time
+	Clone() Image
+}
+
+type image struct {
+	host, org, repo, tag string
+	createdAt            *time.Time `json:",omitempty"`
+}
```

Comment:
Although I could be persuaded, as it's almost a primitive that can be mocked by simply instantiating. But I would argue that you might want to return something which it doesn't return in a normal situation, like an invalid repo.